### PR TITLE
Implement varint

### PR DESCRIPTION
**Pull request by summraznboi**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  "devDependencies": {
+    "@types/jest": "^29.5.12",
+    "@types/lodash": "^4.14.202",
+    "@types/node": "^20.10.7",
+    "@types/promise-retry": "^1.1.6",
+    "jest": "^29.7.0",
+    "ts-jest": "^29.1.2",
+    "typescript": "^5.3.3"
+  },
+  "dependencies": {
```
Comment (mi-yu) [inline package.json:24]: will all of these be needed?

Comment (summraznboi) [reply package.json:23]: good call out, removed unneeded dependencies

File: src/varint.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    while (bigintValue > 0b0111_1111) {
+      bigintValue = bigintValue / 128n - 1n;
+      bufindex--;
+      buffer.writeUInt8(Number(bigintValue & 0xffn) | 0b1000_0000, bufindex);
+    }
+
+    return buffer.subarray(bufindex);
+  }
+
+  private constructor(private value: bigint) {
```
Comment (mi-yu) [inline src/varint.ts:43]: i think this can be `readonly` as well

File: src/varint.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+  saturatingMultiply(other: UInt128): UInt128 {
+    const bigintResult = this.value * other.value;
+    return bigintResult <= UInt128.MAX_VALUE.value
+      ? UInt128.of(bigintResult)
+      : UInt128.MAX_VALUE;
+  }
+
+  divide(other: UInt128): UInt128 {
+    return UInt128.of(this.value / other.value);
```
Comment (mi-yu) [inline src/varint.ts:64]: worth throwing some special error for `other.value === 0`?

Comment (summraznboi) [reply src/varint.ts:64]: Fair enough

**Review by destinymagicchen — APPROVED** [review]

**Review by mi-yu — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "no-content-ord-indexer",
```
Comment (nothing0012) [inline package.json:2]: need a better name here

**Pull request merged**